src/test/java/com/puppycrawl/tools/checkstyle/gui/MainFrameModelTest.java

```diff
@@ -0,0 +1,209 @@
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.gui;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.util.Locale;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.gui.MainFrameModel.ParseMode;
+
+@RunWith(PowerMockRunner.class)
+public class MainFrameModelTest {
+
+    private static final String FILE_NAME_TEST_DATA = "InputJavadocAttributesAndMethods.java";
+    private static final String FILE_NAME_NON_JAVA = "NotJavaFile.notjava";
+    private static final String FILE_NAME_NON_EXISTENT = "non-existent.file";
+    private static final String FILE_NAME_NON_COMPILABLE = "InputIncorrectClass.java";
+
+    private MainFrameModel model;
+    private File testData;
+
+    private static String getPath(String filename) {
+        return "src/test/resources/com/puppycrawl/tools/checkstyle/gui/" + filename;
+    }
+
+    private static String getNonCompilablePath(String filename) {
+        return "src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/" + filename;
+    }
+
+    @Before
+    public void prepareTestData() {
+        model = new MainFrameModel();
+        testData = new File(getPath(FILE_NAME_TEST_DATA));
+    }
+
+    @Test
+    public void testParseModeEnum() {
+        assertEquals("Plain Java", ParseMode.PLAIN_JAVA.toString());
+        assertEquals("Java with comments", ParseMode.JAVA_WITH_COMMENTS.toString());
+        assertEquals("Java with comments and Javadocs",
+                ParseMode.JAVA_WITH_JAVADOC_AND_COMMENTS.toString());
+
+        // ParseMode Enum should only have above 3 enumerations
+        final int expectedEnumerations = 3;
+        assertEquals(expectedEnumerations, ParseMode.values().length);
```

Comment:
In case of enum changes(new item appear) .This assert will only request to update number to new value. If it is expected that some other code of this UT should be extended - so there should be provided appropriate message to define what is expected to be done.